indexer/services/comlink/src/controllers/api/v4/compliance-v2-controller.ts

```diff
@@ -209,9 +211,89 @@ router.post(
         );
       }
 
-      // TODO(OTE-141): Implement logic.
+      /**
+       * If the address doesn't exist in the compliance table:
+       * - if the request is from a restricted country:
+       *  - if the action is ONBOARD, set the status to BLOCKED
+       *  - if the action is CONNECT, set the status to FIRST_STRIKE
+       * - else if the request is not from a non-restricted country:
+       *  - set the status to COMPLIANT
+       *
+       * if the address is COMPLIANT:
+       * - the ONLY action should be CONNECT
```

Comment:
nit: Note that if action is `ONBOARD` it is a no-op.

Reply:
done